### DISCUSSION TITLE
central command-node

### DISCUSSION THREAD

**Discussion by HerrTuulik** [windkh/node-red-contrib-telegrambot#311] [Ideas]
Could it be possible to introduce another node that knows about all the other command nodes and can trigger them based on input? This is something that is desperately needed in systems where the commands, their names and amounts are dynamically generated and deleted, determined by a data-driven architecture.

**Comment by windkh** [comment]
Yes that should not be a problem. This can be integrated into the existing control node.

**Comment by windkh** [comment]
@HerrTuulik Do you want to trigger all commands at the same time? or do you want to send a command string "/foo" into the  node and then it should issue the command node for "/foo"???

**Reply by HerrTuulik** [reply]
The latter. I send a payload with "/foo" into the node and it should trigger the corresponding /foo node whereever it is.

**Comment by windkh** [comment]
BTW sorry for delay, I was on vacation

**Reply by HerrTuulik** [reply]
No problem, i've just recovered from an illness myself.

**Reply by windkh** [reply]
Ok understood

**Comment by windkh** [comment]
@HerrTuulik another question:
normally command node emits a message that contains the original message, the messageId, userid, chatId, ...
without the chatId for example you can not send a response for a command. 

There are two options:
1. you must provide chatId together with your command string "/Foo" when triggering the node: { chatId : 12345, message : "/foo" };
2. you can only provide "/foo" and then the command node will only emit a message with minimum info: you will then not be able to respond to a command in telegram.

Any ideas?

**Comment by windkh** [comment]
command node can now trigger commands. see new example in example folder
